src/Data/List/Ext/Properties.agda

```diff
@@ -54,11 +231,440 @@ module _ {a} {A : Set a} ⦃ _ : DecEq A ⦄ where
           helper : ∀ {l l' a} → a ∈ l ++ l' ⇔ (a ∈ l ⊎ a ∈ l')
           helper = mk⇔ (∈-++⁻ _) Data.Sum.[ ∈-++⁺ˡ , ∈-++⁺ʳ _ ]
 
--- TODO: stdlib?
-AllPairs⇒≡∨R∨Rᵒᵖ : ∀ {ℓ ℓ'} {A : Set ℓ} {R : A → A → Set ℓ'} {a b l}
-                 → AllPairs R l → a ∈ˡ l → b ∈ˡ l → a ≡ b ⊎ R a b ⊎ R b a
-AllPairs⇒≡∨R∨Rᵒᵖ [] = λ ()
-AllPairs⇒≡∨R∨Rᵒᵖ (x ∷ h) (here refl) (here refl) = inj₁ refl
-AllPairs⇒≡∨R∨Rᵒᵖ (x ∷ h) (here refl) (there b∈l) = inj₂ (inj₁ (All.lookup x b∈l))
-AllPairs⇒≡∨R∨Rᵒᵖ (x ∷ h) (there a∈l) (here refl) = inj₂ (inj₂ (All.lookup x a∈l))
-AllPairs⇒≡∨R∨Rᵒᵖ (x ∷ h) (there a∈l) (there b∈l) = AllPairs⇒≡∨R∨Rᵒᵖ h a∈l b∈l
+-----------------------------------------------------------
+-------- properties of concatenation and union ------------
+-----------------------------------------------------------
+module _ {a} {A : Set a} where
```

Comment:
Most things in here are in `Data.List.Relation.Unary.Any.Properties` (note that `_∈_` is defined via `Any`)